catalog/core/catalog-core-camelcomponent/src/main/java/ddf/camel/component/catalog/content/ContentProducerDataAccessObject.java

```diff
@@ -131,22 +131,31 @@ public void createContentItem(FileSystemPersistenceProvider fileIdMap, ContentEn
             Map<String, Object> headers) throws SourceUnavailableException, IngestException {
         LOGGER.debug("Creating content item.");
 
-        String key = String.valueOf(ingestedFile.getAbsolutePath()
-                .hashCode());
-        String id = fileIdMap.loadAllKeys()
-                .contains(key) ? String.valueOf(fileIdMap.loadFromPersistence(key)) : null;
-
+        String refKey = (String) headers.get(Constants.STORE_REFERENCE_KEY);
+        String safeKey = null;
+        String id = null;
+        String name = ingestedFile.getName();
+
+        if (refKey != null) {
```

Comment:
If the file is being ingested into content the store, the reference will be null. References are for files that live outside of the content store. I can add an explanatory comment.